src/UsersEpic.js

```diff
@@ -0,0 +1,43 @@
+//@flow
+import {ofType} from 'redux-observable';
+import {ACTIONS} from 'redux-api-call';
+import {map, filter} from 'rxjs/operators';
+import {addUsers} from './UsersAction';
+import {combineEpics} from 'redux-observable';
+import {fetchUsersAC} from './state.js';
+
+const fetchUsersCompleteEpic = action$ =>
+  action$.pipe(
+    ofType(ACTIONS.COMPLETE),
+    filter(action => action.payload.name === 'FETCH_USERS'),
+    map(({payload}) => {
+      const fetchedUsers = payload.json.results || [];
+      return addUsers(fetchedUsers);
+    }),
+  );
+
+let fetchUsersNetworkFails: number = 0;
+const fetchUsersNetworkFailureEpic = action$ =>
+  action$.pipe(
+    ofType(ACTIONS.FAILURE),
+    filter(
+      ({payload}) =>
+        payload.name === 'FETCH_USERS' &&
+        payload.json.message === 'Network request failed',
+    ),
+    map(res => {
+      if (++fetchUsersNetworkFails < 3) {
+        return fetchUsersAC();
+      } else {
+        fetchUsersNetworkFails = 0;
+        return {type: 'NO_ACTION'};
```

Comment:
Why we need to return NO_ACTION here?

Reply:
My goal here is not to dispatch any action. I did try to return an empty object but received this error "Actions may not have an undefined "type" property". Hence, I put a dummy type here.

Reply:
Please try undefined, or empty array `[]`

Reply:
Both approaches yielded the following error messages.
![image](https://user-images.githubusercontent.com/81946174/117098523-f3b08c00-ad98-11eb-861a-e7f245161c91.png)